src/components/results/ResultTables.js

```diff
@@ -77,42 +126,98 @@ const ResultTables = ({ headerElem }) => {
   }
 
   return order.length > 0 ? (
-    <Stack direction="row" spacing={5}>
-      <Box sx={{ maxHeight: maxHeight, overflow: 'auto' }}>
-        <AutoScrollableMenu
-          links={order.map((qId) => {
-            const data = questionData.get(qId).first();
-            return { label: data.question, link: data.innerLink };
-          })}
-          elemRefs={questionRefs}
-          containerRef={questionContainerRef}
-        />
-      </Box>
-      <Box
-        sx={{
-          overflow: 'auto',
-          width: '100%',
-          height: maxHeight,
-          pr: 1,
-          scrollBehavior: 'smooth',
-        }}
-        ref={questionContainerRef}
-      >
-        {order.map((qId) => (
+    users?.size > 0 ? (
+      <Stack direction="row" spacing={5}>
+        <Box sx={{ maxHeight: maxResultViewHeight }}>
+          <ResultTablesMenu
+            tab={tab}
+            setTab={setTab}
+            tableMenuElem={tableMenuElem}
+          />
+          <Box sx={{ maxHeight: maxHeightScrollableMenu, overflow: 'auto' }}>
+            <TabPanel tab={tab} index={TABLE_BY_QUESTION_PANEL_IDX}>
+              <AutoScrollableMenu
+                links={order.map((qId) => {
+                  const data = questionData.get(qId).first();
```

Comment:
It's probably safer to consider the case where the question might not exist anymore (even tho it's supposed to be in order).

Reply:
Ok, I will add a `null-safe` call